### PR TITLE
Update logic for when Advertisement labels are required

### PR DESCRIPTION
**Pull request by JonNorman**
## What does this change?
In anticipation of some changes upstream to our DFP native styles, this PR checks for specific markers in the creatives to determine whether an Advertisement label should be rendered or not.

Eventually a lot of this logic will be simplified, but because the systems are being updated separately, the old logic and new logic need to remain in place for the time being.

## What is the value of this and can you measure success?
More determined and explicit rules for when an Advertisement label should be rendered, no longer determined from opaque logic around ad sizes and formats.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.js [hunk 1]
```diff
@@ -5,14 +5,29 @@ import popupTemplate from 'raw-loader!commercial/views/ad-feedback-popup.html';
...
+const shouldRenderLabel = adSlotNode => {
+    const explicitlyOptedOut =
+        adSlotNode.getElementsByClassName(`${labelRequiredSuffix}-false`)
+            .length > 0;
+    const explicitlyOptedIn =
+        adSlotNode.getElementsByClassName(`${labelRequiredSuffix}-true`)
+            .length > 0;
+
+    return (
+        (!explicitlyOptedOut || explicitlyOptedIn) &&
```
Comment (LATaylor-guardian) [inline static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.js]: Might be worth assigning this expression to a well named variable and returning that for readability.

**Review by rich-nguyen — APPROVED** [review]
Not sure why the hide changes are here

File: static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.js (hunk 1 above)
Comment (JonNorman) [reply static/src/javascripts/projects/commercial/modules/dfp/render-advert-label.js]: Yeah - I'm planning on ripping this all out in a day or so - the logic is horrendous at the moment.

**Comment by JonNorman** [comment]
@rich-nguyen yep that was when I was going in a different direction on this branch. I'll remove that commit.

**Comment by JonNorman** [comment]
In terms of what the `hide` changes do: it just moves the functionality for hiding an element out of the messenger framework, so it can be used elsewhere. It also simplifies the actual logic as there are some redundant checks in there. As I mentioned above, there's no reason for that to be in this PR anymore, but since it is small and doesn't affect much, I will keep it in.

**Pull request merged**